recipe/meta.yaml

```diff
@@ -10,28 +21,32 @@ source:
   fn: {{ name }}-{{ version }}.tar.xz
   url: http://software.ligo.org/lscsoft/source/lalsuite/{{ name }}-{{ version }}.tar.xz
   sha256: {{ sha256 }}
+  patches:
+    # fix overlinking
+    - pkgconfig.patch
 
 build:
-  number: 2
+  number: {{ build }}
   skip: true  # [win]
+  string: "{{ fft_impl }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}"
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - pkg-config
+    - pkg-config >=0.18.0
     - make
     - bc
     - swig >=3.0.7
   host:
-    - libcblas
```

Comment:
You need to keep this here.

Reply:
Can I ask why this is needed?

Reply:
`lal` needs BLAS right?

Reply:
no, `lal` needs `gsl`, but has _always_ been built against openblas until we started using conda-forge, so I naively put in `openblas` first-time around. The package builds fine without any specification of `blas`, verified by using `--error-overlinking --error-overdepending` (if that's sufficient proof) on the last commit (before your suggestions).

Reply:
There's a stray `cblas_dgemv` in `lalinference/src/bayestar_distance.c`

Reply:
nvm, that's lalinference. There's something linking to openblas somewhere.

Reply:
Where do you see that?

Reply:
Line 4339 in https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=24518&view=logs&jobId=326f6796-80bd-5636-66bd-ae7357ee2d7f&taskId=f73a107e-a87b-54a7-91d9-8b84ed158eec&lineStart=4339&lineEnd=4340&colStart=1&colEnd=1

Reply:
Ah, ok, I'll take a look at that (tomorrow). Thanks again for leading me through this one, I want to get it right and not end up with a substandard recipe.

Reply:
@isuruf, I believe that `lal` is linking directly against `openblas` _only_ because `gsl` is telling it to via `pkgconfig`. Consider the following line in `gsl.pc` from `gsl-2.4-ha2d443c_1005`:

```
Libs: -L/opt/anaconda1anaconda2anaconda3/lib -lgsl ${GSL_CBLAS_LIB}  -lcblas -lm
```

I believe that this is an error on the part of gsl and that this should actually read:

```
Libs: -L/opt/anaconda1anaconda2anaconda3/lib -lgsl ${GSL_CBLAS_LIB}
Libs.private: -L/opt/anaconda1anaconda2anaconda3/lib -lgsl ${GSL_CBLAS_LIB}  -lcblas -lm
```

so as to not force all downstream libraries to link against `libcblas` (which is just a symlink to `libopenblas` when using that implementation).

If that's the case, is it still better to have `lal` directly reference `libcblas` and `openblas` in its requirements, or can I just leave it and hope for an updated gsl that doesn't overlink?